src/insi-client.service.ts

```diff
@@ -79,34 +82,31 @@ export class INSiClient {
    * Fetches INS information of a person
    * @param  {INSiPerson} person the person who's information are about to be fetched
    * @param  {string} requestId of the current request to Ins
-   * @returns Promise<INSiFetchInsResponse>
+   * @returns Promise<INSiFetchInsResponse[]>
    */
-  public async fetchIns(person: INSiPerson, { requestId = uuidv4() } = {}): Promise<INSiFetchInsResponse> {
+  public async fetchIns(person: INSiPerson, { requestId = uuidv4() } = {}): Promise<INSiFetchInsResponse[]> {
     if (!this._soapClient) {
       throw new Error('fetchIns ERROR: you must init client security first');
     }
     return this._launchSoapRequestForPerson(person, requestId);
   }
 
-  private async _launchSoapRequestForPerson(person: INSiPerson, requestId: string): Promise<INSiFetchInsResponse> {
-    let rawSoapResponse;
-    const failedRequests = [];
+  private async _launchSoapRequestForPerson(person: INSiPerson, requestId: string): Promise<INSiFetchInsResponse[]> {
+    const responses: INSiFetchInsResponse[] = [];
     const namesToSendRequestFor = person.getSoapBodyAsJson();
-    const { method } = INSiSoapActions[INSiSoapActionsName.FETCH_FROM_IDENTITY_TRAITS];
     const savedOverriddenHttpClientResponseHandler = this._httpClient.handleResponse;
     for (let i = 0; i < namesToSendRequestFor.length; i++) {
```

Comment:
Because that's a requirement for the INS service. You need to do this in order to be INSi compliant. In consequence we chose with abdallah to have this logique here.
This way the backend handle a person with multiple name and dont bother with the detail.